### PR TITLE
Add chef dependency bumping to bot

### PR DESCRIPTION
**Pull request by jkeiser**
use git checkout .bundle/config to ensure we don't modify bundler config

### CODE REVIEW

File: lib/lita/handlers/dependency_updater.rb [hunk 1]
```diff
@@ -38,9 +38,17 @@ class DependencyUpdater < Handler
         chefdk: {
           pipeline: "chefdk-trigger-release",
           github_url: "git@github.com:chef/chef-dk.git",
-          version_bump_command: "bundle install && bundle exec rake version:bump",
+          version_bump_command: "bundle install && bundle exec rake version:bump && git checkout .bundle/config",
```
Comment (danielsdeleo) [inline lib/lita/handlers/dependency_updater.rb]: You can probably remove this, I think I just copy-pasta'd from the other one and forgot to take it out. Alternatively, could merge everything into one project config file.

File: lib/lita/handlers/dependency_updater.rb [hunk 2]
```diff
@@ -38,9 +38,17 @@ class DependencyUpdater < Handler
...
           version_show_command: "bundle exec rake version:show",
-          dependency_update_command: "bundle install && bundle exec rake dependencies",
+          dependency_update_command: "bundle install && bundle exec rake dependencies && git checkout .bundle/config",
+          inform_channel: "workflow-pool"
+        },
+        chef: {
+          pipeline: "chef-trigger-release",
+          github_url: "git@github.com:chef/chef.git",
+          version_bump_command: "bundle install && bundle exec rake version:bump && git checkout .bundle/config",
```
Comment (danielsdeleo) [inline lib/lita/handlers/dependency_updater.rb]: same as above.

**Comment by jkeiser** [comment]
I'm unclear whether this is working; can't get debug info out of it. I'm adding some logging stuff here to see what I can see.

File: lita_config.rb.example [hunk 3]
```diff
@@ -19,4 +19,32 @@ Lita.configure do |config|
...
+      version_bump_command: "bundle install && bundle exec rake version:bump && git checkout .bundle/config",
+      version_show_command: "bundle exec rake version:show",
+      dependency_update_command: "bundle install && bundle exec rake dependencies && git checkout .bundle/config",
+      inform_channel: "workflow-pool"
+    },
+    "chefdk" => {
+      pipeline: "chefdk-trigger-release",
+      github_url: "git@github.com:chef/chef-dk.git",
+      version_bump_command: "bundle install && bundle exec rake version:bump && git checkout .bundle/config",
+      version_show_command: "bundle exec rake version:show",
```
Comment (danielsdeleo) [inline lita_config.rb.example]: It probably makes more sense to have a convention for what these scripts are called and then put them in the projects

Comment (danielsdeleo) [inline lita_config.rb.example]: e.g., `PROJECT_ROOT/botscripts/version-bump.sh` or whatever

Comment (jkeiser) [inline lita_config.rb.example]: Good idea, I'll add that.

**Pull request merged**